video/out/gpu/video.c

```diff
@@ -1168,34 +1172,41 @@ static struct mp_pass_perf render_pass_quad(struct gl_video *p,
     gl_transform_vec(t, &x[1], &y[1]);
 
     for (int n = 0; n < 4; n++) {
-        struct vertex *v = &va[n];
-        v->position.x = x[n / 2];
-        v->position.y = y[n % 2];
-        for (int i = 0; i < p->pass_img_num; i++) {
-            struct image *s = &p->pass_img[i];
+        struct vertex_pt *vs = &p->tmp_vertex[num_vertex_attribs * n];
+        // vec2 position in idx 0
+        vs[0].x = x[n / 2];
+        vs[0].y = y[n % 2];
+        for (int i = 0; i < p->num_pass_imgs; i++) {
+            struct image *s = &p->pass_imgs[i];
             if (!s->tex)
                 continue;
             struct gl_transform tr = s->transform;
             float tx = (n / 2) * s->w;
             float ty = (n % 2) * s->h;
             gl_transform_vec(tr, &tx, &ty);
             bool rect = s->tex->params.non_normalized;
-            v->texcoord[i].x = tx / (rect ? 1 : s->tex->params.w);
-            v->texcoord[i].y = ty / (rect ? 1 : s->tex->params.h);
+            // vec2 texcoordN in idx N+1
+            vs[i + 1].x = tx / (rect ? 1 : s->tex->params.w);
+            vs[i + 1].y = ty / (rect ? 1 : s->tex->params.h);
         }
     }
 
-    va[4] = va[2];
-    va[5] = va[1];
+    memmove(&p->tmp_vertex[num_vertex_attribs * 4],
+            &p->tmp_vertex[num_vertex_attribs * 2],
+            vertex_stride);
```

Comment:
Does this actually make sense? Is vertex_stride == num_vertex_attrib*sizeof(tmp_vertex[0])?

Reply:
From earlier in the function:

```
    size_t vertex_stride = num_vertex_attribs * sizeof(struct vertex_pt);
```

and `tmp_vertex` is a `struct vertex_pt *`.

Reply:
Oh, OK. That would be trouble-some if we ever needed other vertex attributes than vertex_pts, but should work for now.

Reply:
My thoughts exactly